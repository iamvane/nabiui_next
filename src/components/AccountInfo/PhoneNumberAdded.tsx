import * as React from 'react';

import {
  Grid,
  IconButton,
  Typography
} from '@material-ui/core';
// import Done from '@material-ui/icons/Done';
import dynamic from "next/dynamic";
const Done = dynamic(() => import('@material-ui/icons/Done'), {
  ssr: false,
});

import { PhoneValidationFormComponent } from './constants';

interface Props {
  phoneNumber: string | undefined;
  isEditing?: boolean;
}

const PhoneNumberAdded: React.StatelessComponent<Props> = props => {
  let value = props.phoneNumber + '';
  let formattedText = value.replace(/^(\d{3})(\d{3})(\d{4}).*/, '+1 ($1) $2-$3');

  return (
    <Grid item={true} xs={12} container={true}>
      <Grid xs={12} md={props.isEditing ? 5 : null} item={true}>
        <Typography>
          {formattedText}
        </Typography>
      </Grid>
      <Grid xs={12} md={props.isEditing ? 6 : null} item={true}>
        <Typography
          color="primary"
          className="nabi-margin-right-xsmall nabi-color-nabi nabi-display-inline"
        >
          {PhoneValidationFormComponent.validationSucessful}
        </Typography>
        <IconButton color="primary" disabled={true}>
          <Done />
        </IconButton>
      </Grid>
    </Grid>
  );
};

export default PhoneNumberAdded;
