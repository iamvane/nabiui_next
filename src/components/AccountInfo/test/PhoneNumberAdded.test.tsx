import * as React from 'react';

import { shallow } from 'enzyme';
import PhoneNumberAdded from 'components/AccountInfo/PhoneNumberAdded';

describe('PhoneNumberAdded', () => {
  let wrapper: any;

  beforeEach(() => {
    wrapper = shallow(
      <PhoneNumberAdded
        phoneNumber="232312313"
      />
    );
  });

  it('Renders the component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
