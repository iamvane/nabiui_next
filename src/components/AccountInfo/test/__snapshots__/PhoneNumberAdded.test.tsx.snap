// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhoneNumberAdded Renders the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PhoneNumberAdded
    phoneNumber="232312313"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <WithStyles(Grid)
          item={true}
          md={undefined}
          xs={12}
        >
          <WithStyles(Typography)>
            232312313
          </WithStyles(Typography)>
        </WithStyles(Grid)>,
        <WithStyles(Grid)
          item={true}
          md={undefined}
          xs={12}
        >
          <WithStyles(Typography)
            className="nabi-margin-right-xsmall nabi-color-nabi nabi-display-inline"
            color="primary"
          >
            Phone Verified
          </WithStyles(Typography)>
          <WithStyles(IconButton)
            color="primary"
            disabled={true}
          >
            <WithStyles(Icon)>
              done
            </WithStyles(Icon)>
          </WithStyles(IconButton)>
        </WithStyles(Grid)>,
      ],
      "container": true,
      "item": true,
      "xs": 12,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <WithStyles(Typography)>
            232312313
          </WithStyles(Typography)>,
          "item": true,
          "md": undefined,
          "xs": 12,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "232312313",
          },
          "ref": null,
          "rendered": "232312313",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(Typography)
              className="nabi-margin-right-xsmall nabi-color-nabi nabi-display-inline"
              color="primary"
            >
              Phone Verified
            </WithStyles(Typography)>,
            <WithStyles(IconButton)
              color="primary"
              disabled={true}
            >
              <WithStyles(Icon)>
                done
              </WithStyles(Icon)>
            </WithStyles(IconButton)>,
          ],
          "item": true,
          "md": undefined,
          "xs": 12,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Phone Verified",
              "className": "nabi-margin-right-xsmall nabi-color-nabi nabi-display-inline",
              "color": "primary",
            },
            "ref": null,
            "rendered": "Phone Verified",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(Icon)>
                done
              </WithStyles(Icon)>,
              "color": "primary",
              "disabled": true,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "done",
              },
              "ref": null,
              "rendered": "done",
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <WithStyles(Grid)
            item={true}
            md={undefined}
            xs={12}
          >
            <WithStyles(Typography)>
              232312313
            </WithStyles(Typography)>
          </WithStyles(Grid)>,
          <WithStyles(Grid)
            item={true}
            md={undefined}
            xs={12}
          >
            <WithStyles(Typography)
              className="nabi-margin-right-xsmall nabi-color-nabi nabi-display-inline"
              color="primary"
            >
              Phone Verified
            </WithStyles(Typography)>
            <WithStyles(IconButton)
              color="primary"
              disabled={true}
            >
              <WithStyles(Icon)>
                done
              </WithStyles(Icon)>
            </WithStyles(IconButton)>
          </WithStyles(Grid)>,
        ],
        "container": true,
        "item": true,
        "xs": 12,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(Typography)>
              232312313
            </WithStyles(Typography)>,
            "item": true,
            "md": undefined,
            "xs": 12,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "232312313",
            },
            "ref": null,
            "rendered": "232312313",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(Typography)
                className="nabi-margin-right-xsmall nabi-color-nabi nabi-display-inline"
                color="primary"
              >
                Phone Verified
              </WithStyles(Typography)>,
              <WithStyles(IconButton)
                color="primary"
                disabled={true}
              >
                <WithStyles(Icon)>
                  done
                </WithStyles(Icon)>
              </WithStyles(IconButton)>,
            ],
            "item": true,
            "md": undefined,
            "xs": 12,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Phone Verified",
                "className": "nabi-margin-right-xsmall nabi-color-nabi nabi-display-inline",
                "color": "primary",
              },
              "ref": null,
              "rendered": "Phone Verified",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <WithStyles(Icon)>
                  done
                </WithStyles(Icon)>,
                "color": "primary",
                "disabled": true,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "done",
                },
                "ref": null,
                "rendered": "done",
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
